File: Programmers/Lv3/가장 먼 노드.js
function solution(n, edge) {
  // start = 1
  var answer = 0;
  var graph = Array.from(Array(n + 1), () => Array());

  for (v of edge) {
    let a = v[0];
    let b = v[1];
    graph[a].push(b);
    graph[b].push(a);
  }

  var visited = new Set();
  var dist = new Array(n + 1).fill(0);
  const bfs = () => {
    var q = [1];
    visited.add(1);

    while (q.length > 0) {
      let node = q.shift();

      graph[node].forEach((v) => {
        if (!visited.has(v)) {
          q.push(v);
          visited.add(v);
          dist[v] = dist[node] + 1;
        }
      });
    }
  };

  bfs();
  const maxVal = Math.max(...dist);
  for (num of dist) {
    if (num == maxVal) {
      answer += 1;
    }
  }
  return answer;
}

File: Programmers/Lv3/가장 먼 노드.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(int n, vector<vector<int>> edge) {
    // 시작은 1번
    unordered_map<int, vector<int>> graph;
    for(auto& e : edge){
        graph[e[0]].push_back(e[1]);
        graph[e[1]].push_back(e[0]);
    }
    
    queue<int> q;
    vector<int> visited(n + 1, 0);
    
    q.push(1);
    visited[1] = 1;
    
    while(!q.empty()){
        int node = q.front();
        q.pop();
        
        for(int& adj : graph[node]){
            if(visited[adj] == 0){
                q.push(adj);
                visited[adj] = visited[node] + 1;
            }
        }
    }
    
    vector<int> sliced(visited.begin() + 1, visited.end());
    
    int maxVal = *max_element(sliced.begin(), sliced.end());
    
    int cnt = count(sliced.begin(), sliced.end(), maxVal);
    
    return cnt;
}